### PR TITLE
feat: migrate to @react-google-maps/api

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** `Feature` ⭐ 

**Description** @appbaseio/reactive-maps lib would now use the `@react-google-maps/api`, which is a maintained rewrite of `react-google-maps`.

- [x] Migrate to @react-google-maps/api
- [x] Use mapKey in ReactiveBase to load map scripts

[📔  Notion](https://www.notion.so/appbase/ReactiveMaps-v3-Stable-Release-825feebb39a94dcba260c9b72dd11af1)

https://www.loom.com/share/b4c8d44503a04f96aad5bee105ed8bcd
https://www.loom.com/share/1845099fbbbb41d085deadaa0fdd4c7f

**Related PR(s)**
1. https://github.com/appbaseio/reactivecore/pull/122

### CODE REVIEW

File: packages/maps/src/components/result/GoogleMapMarker.js [hunk 1]
```diff
@@ -164,15 +159,15 @@ class GoogleMapMarker extends React.Component {
 				return (
 					<MarkerWithLabel
 						key={marker._id}
-						labelAnchor={new window.google.maps.Point(0, 30)}
+						labelAnchor={new window.google.maps.Point(0, 0)}
```
Comment (bietkul) [inline packages/maps/src/components/result/GoogleMapMarker.js:162]: Is there any reason for the change?

File: packages/maps/src/components/result/addons/components/ScriptLoader.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	if (
+		!window.google
+		&& window.GOOGLE_SCRIPT_LOCK_ACQUIRED === true
+		&& !mapScriptLoadStatus.loaded
+	) {
+		return 'Initializing...';
+	}
+
+	if (!window.google && typeof window.GOOGLE_SCRIPT_LOCK_ACQUIRED === 'undefined') {
```
Comment (bietkul) [inline packages/maps/src/components/result/addons/components/ScriptLoader.js:39]: @mohdashraf010897 We discussed to use Redux actions for it. You're just using variables here, which is incorrect. It won't re-render the component once the script is loaded.

File: packages/maps/src/components/result/addons/components/ScriptLoader.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ScriptLoader.defaultProps = {
+	mapScriptLoadStatus: {},
+};
+const mapStateToProps = state => ({
+	mapKey: state.config.mapKey,
+	mapScriptLoadStatus: state.googleMapScriptStatus,
+});
+
+const mapDispatchtoProps = dispatch => ({
+	setMapScriptLoading: loading => dispatch({ type: SET_GOOGLE_MAP_SCRIPT_LOADING, loading }),
```
Comment (bietkul) [inline packages/maps/src/components/result/addons/components/ScriptLoader.js]: Define actions in `reactivecore`.

File: packages/maps/src/components/result/addons/components/helper.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+/* global google */
+function reduce(obj, fn, acc) {
+	return Object.keys(obj).reduce((newAcc, key) => fn(newAcc, obj[key], key), acc);
+}
+
+function forEach(obj, fn) {
```
Comment (bietkul) [inline packages/maps/src/components/result/addons/components/helper.js:6]: Are these polyfills?

**Review by bietkul — CHANGES_REQUESTED** [review]

**Comment by bietkul** [comment]
@mohdashraf010897 
1.  Fix conflicts
2. Resolve comments
3. Where is the docs link?

File: packages/maps/src/components/result/addons/components/helper.js (hunk 4 above)
Comment (mohdashraf010897) [reply packages/maps/src/components/result/addons/components/helper.js:6]: @bietkul No, these are not polyfills. Just some generalized wrapper around the function.

File: packages/maps/src/components/result/addons/components/ScriptLoader.js (hunk 3 above)
Comment (mohdashraf010897) [reply packages/maps/src/components/result/addons/components/ScriptLoader.js]: @bietkul Have already done that here https://github.com/appbaseio/reactivecore/pull/122

File: packages/maps/src/components/result/addons/components/ScriptLoader.js (hunk 2 above)
Comment (mohdashraf010897) [reply packages/maps/src/components/result/addons/components/ScriptLoader.js:39]: @bietkul redux actions alone are not enough. Think about the first render of the components(say ReactiveGoogleMap and GeoDistanceDropdown are used in one app together), at the initial render they both would be having the same value of the redux state, only the rerender would supply them with the updated value, and till then they both would have requested the google script individually.

**Comment by bietkul** [comment]
1. Label anchor: Visualization changes after updating the offset value. Provide screenshots.
2. Add check for `window` object
3. Use redux actions from reactivecore 
4. Define `mapLibraries` prop in ReactiveBase (Update docs and migration guide)

File: packages/maps/src/components/result/GoogleMapMarker.js (hunk 1 above)
Comment (mohdashraf010897) [reply packages/maps/src/components/result/GoogleMapMarker.js:162]: @bietkul 
https://www.loom.com/share/0a95d9227a43443397551d8df009dbdc

**Comment by mohdashraf010897** [comment]
> 1. Label anchor: Visualization changes after updating the offset value. Provide screenshots.
> 2. Add check for `window` object
> 3. Use redux actions from reactivecore
> 4. Define `mapLibraries` prop in ReactiveBase (Update docs and migration guide)

1. https://www.loom.com/share/0a95d9227a43443397551d8df009dbdc
2,3,4 ---> https://www.loom.com/share/ee57020ec631465a806483f86c2645fb

Have resolved conflicts too

**Review by bietkul — APPROVED** [review]

**Review by bietkul — APPROVED** [review]

**Pull request merged**